unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,235 @@
+            {"register": {"transaction_id": TRANSACTION_ID}})
+        assert redis_client_auto_register._check_buffer_message_matching_command_and_id(
+            data) is None
+
+    @staticmethod
+    def test_check_buffer_message_matching_command_and_id_throws_exception(
+            redis_client_auto_register):
+        """Checks if _check_buffer_message_matching_command_and_id throws
+           exception if the message does not contain transaction_id."""
+        message = {}
+        with pytest.raises(RedisAPIException,
+                           match="The answer message does not contain a valid "
+                                 "'transaction_id' member."):
+            redis_client_auto_register._check_buffer_message_matching_command_and_id(message)
+
+    @staticmethod
+    def test_check_buffer_message_matching_command_and_id_throws_another_exception(
+            redis_client_auto_register):
+        """Check if the buffer message has no matching command response throws an exception."""
+        data = {"transaction_id": TRANSACTION_ID}
+        with pytest.raises(RedisAPIException,
+                           match="There is no matching command response"
+                                 " in _blocking_command_responses."):
+            redis_client_auto_register._check_buffer_message_matching_command_and_id(data)
+
+    @staticmethod
+    def test_register(redis_client_auto_register):
+        """Check if the redis.db.publish is called with correct arguments."""
+        redis_client_auto_register.redis_db.publish = MagicMock()
+        redis_client_auto_register.register(is_blocking=False)
+        transaction_id = redis_client_auto_register._blocking_command_responses[
+            "register"]["transaction_id"]
+        data = {"name": AREA_ID, "transaction_id": transaction_id}
+        redis_client_auto_register.redis_db.publish.assert_called_once_with(
+            f"{AREA_ID}/register_participant", json.dumps(data))
+
+    @staticmethod
+    @patch("d3a_api_client.redis_client_base.wait_until_timeout_blocking",
+           side_effect=AssertionError)
+    def test_register_self_active_false_throws_exception(mock_wait_until_timeout_blocking,
+                                                         redis_client_auto_register):
+        """Check whether if is active is false throws exception."""
+        with pytest.raises(RedisAPIException,
+                           match="API registration process timed out. "
+                                 "Server will continue processing "
+                                 "your request on the background "
+                                 "and will notify you as soon as the "
+                                 "registration has been completed."):
+            redis_client_auto_register.register(is_blocking=True)
+
+    @staticmethod
+    def test_register_self_active_true_throw_exception(redis_client_auto_register):
```

Comment:
Nit: for this test to be complete we could also check that:
- `self.redis_db.publish` is not called;
- `wait_until_timeout_blocking` is not called.

Reply:
I will change these.